### PR TITLE
[examples] Solving the diffusion issue of incompatibility issue#3169

### PR DESCRIPTION
**Pull request by NatalieC323**
Updated GitHub documentation: 
1. requirements.txt content is incompatible with the version and cannot be downloaded. It is recommended to change the content opencv-python==4.6.0 to opencv-python==4.6.0.66 (otherwise an error will be reported saying that the corresponding version cannot be found). Similarly, torchmetrics==0.6 should be changed into torchmetrics==0.7 (otherwise it will raise an error of incompatibility)

2. The instructions 'pip install colossalai' and 'pip install xformer' given in Readme will forward to the latest version. To download the suitable version for this training, pip install colossalai==0.2.5, pip install xformer==0.12.0 is recommended instead

3. Cuda version installation guide is added to Readme. The recommended version is cuda11.6 to 11.8.

### CODE REVIEW

File: examples/images/diffusion/README.md [hunk 1]
```diff
@@ -57,32 +56,38 @@ pip install transformers diffusers invisible-watermark
...
-##### From Source
+##### From Source:
 ```
 git clone https://github.com/Lightning-AI/lightning.git
 pip install -r requirements.txt
 python setup.py install
 ```
 
-##### From pip
+##### From pip:
```
Comment (Fazziekey) [inline examples/images/diffusion/README.md:68]: remove ":"

File: examples/images/diffusion/environment.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ channels:
   - pytorch
   - defaults
 dependencies:
-  - python=3.9.12
+  - python=3.8.16
```
Comment (Fazziekey) [inline examples/images/diffusion/environment.yaml:6]: use 3.9.12

**Comment by binmakeswell** [comment]
Please resolve the conflicting files, thanks.
examples/images/diffusion/README.md

**Comment by Fazziekey** [comment]
the PR looks good, but it still have some conflict, please run git pull origin main --rebase

**Review by NatalieC323 — COMMENTED** [review]
I have deleted all ":"s in the Readme file and the python version has been changed back to 3.9.12

**Review by Fazziekey — APPROVED** [review]

**Pull request merged**